File: 02 tests/ex2_str_concat.js
// run deno 02_str_concat.js

/* 02_str_concat.js

Concatenate 10^7 random strings of length 10

:author:	wolf
:created:	2025.05.22
*/

const iterations = 3;
const timing_results = [];

// define the test function
// --- function to be timed-- -

function jsStringConcat(n) {
    let result = "";
    const chars = "abcdefghijklmnopqrstuvwxyz";
    for (let i = 0; i < n; i++) {
        let str = "";
        for (let j = 0; j < 10; j++) {
            str += chars[Math.floor(Math.random() * chars.length)];
        }
        result += str;
    }
    return result;
}


for (let i = 0; i < iterations; i++) {
    const start = performance.now();

    // call the test function
    jsStringConcat(10 ** 7);

    const end = performance.now();

    timing_results.push(end - start);
}


console.log("Elapsed time: ",
    timing_results.reduce((a, b) => a + b) / iterations / 1000,
    "seconds");

File: 02 tests/ex2_str_concat.py
#!/usr/bin/env python3

""" ex2_str_concat.py

Concatenate 10^7 random strings of length 10

:author:	wolf
:created:	2025.05.22
"""

import time
import random
import string

iterations = 3
timing_results = []

# define the test function

def python_string_concat(n):
    result = ""
    for _ in range(n):
        result += ''.join(random.choice(string.ascii_lowercase) for _ in range(10))
    return result


for i in range(iterations):
    start = time.time()

    # call the test function
    python_string_concat(10**7)

    end = time.time()

    timing_results.append(end - start)


print(f"Elapsed time:  {sum(timing_results)/iterations:.4f} seconds")
